src/videoCache.js

```diff
@@ -42,17 +43,22 @@ const ttlBufferInSeconds = 15;
  * @param {string} impUrl An impression tracker URL for the delivery of the video ad
  * @return A VAST URL which loads XML from the given URI.
  */
-function wrapURI(uri, impUrl) {
+function wrapURI(uri, impUrl, vastTrackers) {
```

Comment:
This function is now tightly coupled with `vastTrackers`. It should be decoupled. It shouldn't matter if the impression URLs come from vastTrackers; if tomorrow another mechanism is created to obtain impression URLs, this function has to change to accommodate for it. That should not be the case.

I suggest instead, making the argument an array of `impUrls`.
```suggestion
function wrapURI(uri, impUrls) {
```

Reply:
I've changes `wrapURI` but instead of removing `vastTrackers` I removed `impUrl`. The reason is that my goal is to later add support for other vast tracking event such as completion for example. So keeping `vastTrackers` makes for sense. To do so I've made a new function in `src/video.js` which is called `addImpUrlToTrackers` in order to push the vastImpUrl into the trackers map.